.github/workflows/docker.yml

```diff
@@ -0,0 +1,45 @@
+name: Build and Publish Docker Image
+
+on:
+  push:
```

Comment:
As wolftastic mentioned on discord before, push will probably give too many images. I think only doing master and dev for push is better.

`workflow_dispatch` will allow for the escape hatch in case a non-prod version needs to be ran on the infra.

Reply:
In that case, should we only use a SHA tag for images built on branches other than master and dev? Seems like they are necessary only as a fallback if latest or edge does not apply.